crates/router/src/connector/noon/transformers.rs

```diff
@@ -142,6 +143,7 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for NoonPaymentsRequest {
             currency,
             channel: NoonChannels::Web,
             category,
+            reference: item.payment_id.clone(),
```

Comment:
why are we sending the payment_id here,  use `item.attempt_id

Reply:
As discussed with Shakthi, we are passing this payment_id from Euler directly, so that it can be used as a reference during webhooks.